content/en/blog/_posts/XX-userns-enabled-by-default/index.md

```diff
@@ -0,0 +1,280 @@
+greatly protects what host files it can read/write, which process it can send
+signals to, etc. Furthermore, capabilities granted are only valid inside the
+user namespace and not on the host, limiting the impact a container
+escape can have.
+
+ * _Enablement of new use cases_: User namespaces allow containers to gain
+certain capabilities inside their own user namespace without affecting the host.
+This unlocks new possibilities, such as running applications that require
+privileged operations without granting full root access on the host. This is
+particularly useful for running nested containers.
+
+{{< figure src="/images/blog/2024-04-22-userns-beta/image.svg" alt="Image showing IDs 0-65535 are reserved to the host, pods use higher IDs" title="User namespace IDs allocation" class="diagram-medium" >}}
+
+Without using a user namespace, a container running as root in the case of a
+container breakout has root privileges on the node. If some capabilities
+were granted to the container, the capabilities are valid on the host too. None
+of this is true when using user namespaces (modulo bugs, of course 🙂).
+
+## Demos
+
+Rodrigo created demos to understand how some CVEs are mitigated when user
+namespaces are used. We showed them here before, but take a look if you haven't:
+
+Mitigation of CVE 2024-21626 with user namespaces:
+
+{{< youtube id="07y5bl5UDdA" title="Mitigation of CVE-2024-21626 on Kubernetes by enabling User Namespace support" class="youtube-quote-sm" >}}
+
+Mitigation of CVE 2022-0492 with user namespaces:
+
+{{< youtube id="M4a2b4KkXN8" title="Mitigation of CVE-2022-0492 on Kubernetes by enabling User Namespace support" class="youtube-quote-sm" >}}
+
+## Everything you wanted to know about user namespaces in kubernetes
+
+Here we try to answer some of the questions we have been asked about user
+namespaces support in Kubernetes.
+
+**1. What are the requirements to use it?**
+
+The requirements are documented [here][userns-req]. But we will elaborate a bit
+more, in the following questions.
+
+[userns-req]: /docs/concepts/workloads/pods/user-namespaces/#before-you-begin
+
+**2. What is idmap mounts and why the file-systems used need to support it?**
+
+Idmap mounts is a Linux kernel feature that uses a mapping of UID/GIDs when
+accessing a mount. When combined with user namespaces, it greatly simplifies the
+support for volumes, as you can forget about the host UID/GIDs the user
+namespace is using.
+
+In particular, thanks to idmap mounts we can:
+ * Run each pod with different UIDs/GIDs on the host. This is key for the
+   lateral movement prevention we mentioned earlier.
+ * Share volumes with pods that don't use user namespaces.
+ * Enable/disable user namespaces without needing to chown the pod's volumes.
+
+However, support for idmap mounts in the kernel is not black/white. The support
+is per file-system and support was added over time to different file-systems.
```

Comment:
I find this sentence a bit redundant, but am not sure what this sentence actually meant... Something like following?

```suggestion
However, support for idmap mounts in the kernel can be confusing. It
has been added over time to different file-systems, and their maturity can differ.
```

Reply:
No. Their maturity doesn't differ and it's not confusing.

My point here is: is **not** that a given kernel version supports idmap mounts or not. A given kernel version **and** file-system can support idmap mounts or not. Support is per file-system.

Reply:
I've just changed it to sentence that goes straight to the point:

```
Support for idmap mounts in the kernel is per file-system and different kernel releases added support for idmap mounts on different file-systems.
```

I hope this is clear for you. Otherwise please feel free to re-open.

Reply:
Thanks that reads better indeed!